lib/spack/spack/stage.py

```diff
@@ -100,9 +109,17 @@ def create_stage_root(path):
         else:
             p_stat = os.stat(p)
 
-        if user_uid != p_stat.st_uid:
+        if _platform != "win32":
+            owner_uid = p_stat.st_uid
+        else:
+            sid = win32security.GetFileSecurity(
+                p, win32security.OWNER_SECURITY_INFORMATION) \
+                .GetSecurityDescriptorOwner()
+            owner_uid = win32security.LookupAccountSid(None, sid)[0]
```

Comment:
IMO it would be good to wrap this user ID logic into a function that is called here vs. doing Windows-specific handling in this function

Reply:
Agreed, will add to the current refactor based on suggested changes.